pkg/deploy/saveversion.go

```diff
@@ -6,36 +6,21 @@ package deploy
 import (
 	"context"
 	"fmt"
-	"time"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
-	mgmtstorage "github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2021-09-01/storage"
-	"github.com/Azure/go-autorest/autorest/date"
 
 	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/azblob"
 	"github.com/Azure/ARO-RP/pkg/util/pointerutils"
 )
 
 // SaveVersion for current location in shared storage account for environment
-func (d *deployer) SaveVersion(ctx context.Context) error {
+func (d *deployer) SaveVersion(ctx context.Context, tokenCredential azcore.TokenCredential) error {
 	d.log.Printf("saving RP version %s deployed in %s to storage account %s", d.version, d.config.Location, *d.config.Configuration.RPVersionStorageAccountName)
-	t := time.Now().UTC().Truncate(time.Second)
-	res, err := d.globalaccounts.ListAccountSAS(
-		ctx, *d.config.Configuration.GlobalResourceGroupName, *d.config.Configuration.RPVersionStorageAccountName, mgmtstorage.AccountSasParameters{
-			Services:               mgmtstorage.ServicesB,
-			ResourceTypes:          mgmtstorage.SignedResourceTypesO + mgmtstorage.SignedResourceTypesS,
-			Permissions:            mgmtstorage.PermissionsC + mgmtstorage.PermissionsW, // create and write
-			Protocols:              mgmtstorage.HTTPProtocolHTTPS,
-			SharedAccessStartTime:  &date.Time{Time: t},
-			SharedAccessExpiryTime: &date.Time{Time: t.Add(24 * time.Hour)},
-		})
-	if err != nil {
-		return err
-	}
 
-	d.log.Infof("instantiating blobs client using SAS token")
-	sasUrl := fmt.Sprintf("https://%s.blob.%s/?%s", *d.config.Configuration.RPVersionStorageAccountName, d.env.Environment().StorageEndpointSuffix, *res.AccountSasToken)
-	blobsClient, err := azblob.NewBlobsClientUsingSAS(sasUrl, d.env.Environment().ArmClientOptions())
+	d.log.Infof("instantiating blobs client using SAS token for ensure static web content is enabled")
+	serviceUrl := fmt.Sprintf("https://%s.blob.%s", *d.config.Configuration.RPVersionStorageAccountName, d.env.Environment().StorageEndpointSuffix)
+	blobsClient, err := azblob.NewBlobsClientUsingEntra(serviceUrl, tokenCredential, d.env.Environment().ArmClientOptions())
```

Comment:
Maybe instead of constructing the blobsClient in the `SaveVersion` function, we should construct it in the deployer similar to what we do for `clusterKeyvault`, from a consistency perspective.  Then we don't have to pass a token everywhere.  

Thoughts?  Non-blocking, just a comment.

Reply:
Same would apply for uploadBlobsClient below as well.